### PR TITLE
feat: auto-release based on version in Cargo.toml

### PR DESCRIPTION
**Pull request by rswanson**
### TL;DR
Added a new reusable GitHub Action workflow for automatically creating releases based on Rust crate versions.

### What changed?
- Created a new workflow `auto-release-rust.yml` that compares Cargo.toml version against Git tags
  - Workflow automatically creates GitHub releases when crate version differs from latest tag
- Added documentation explaining workflow usage and parameters
- Updated example workflow to demonstrate implementation

### How to test?
1. Add workflow to your Rust project's CD:
```yml
create-release:
  uses: init4tech/actions/.github/workflows/auto-release-rust.yml@main
  with:
    binary-name: "your-crate-name"
```
2. Update version in Cargo.toml
3. Push changes to trigger workflow
4. Verify new release is created with correct version tag

### CODE REVIEW

**Comment by rswanson** [comment]
* **#53** <a href="https://app.graphite.dev/github/pr/init4tech/actions/53?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rswanson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Evalir — APPROVED** [review]
